analysis/analysis.go

```diff
@@ -483,6 +482,37 @@ func assessMetricStatus(metric v1alpha1.Metric, result v1alpha1.MetricResult, te
 	return v1alpha1.AnalysisPhaseRunning
 }
 
+func assessMetricFailureInconclusiveOrError(metric v1alpha1.Metric, result v1alpha1.MetricResult) (v1alpha1.AnalysisPhase, []string) {
+	var msg1, msg2 string
+	var phase v1alpha1.AnalysisPhase
+	if result.Failed > metric.FailureLimit {
+		phase = v1alpha1.AnalysisPhaseFailed
+		msg1 = fmt.Sprintf("assessed %s", phase)
+		msg2 = fmt.Sprintf("failed (%d) > failureLimit (%d)", result.Failed, metric.FailureLimit)
+		message := []string{msg1, msg2}
+		return phase, message
+	}
+	if result.Inconclusive > metric.InconclusiveLimit {
+		phase = v1alpha1.AnalysisPhaseInconclusive
+		msg1 = fmt.Sprintf("assessed %s", phase)
+		msg2 = fmt.Sprintf("inconclusive (%d) > inconclusiveLimit (%d)", result.Inconclusive, metric.InconclusiveLimit)
+		message := []string{msg1, msg2}
+		return phase, message
+	}
+	consecutiveErrorLimit := DefaultConsecutiveErrorLimit
+	if metric.ConsecutiveErrorLimit != nil {
+		consecutiveErrorLimit = *metric.ConsecutiveErrorLimit
+	}
```

Comment:
I know this was here before your change, but can you give this into the utils/defaults package and create a method called `GetConsecutiveErrorLimitOrDefault` with this logic?